src/transport/grpc/mod.rs

```diff
@@ -73,11 +79,28 @@ pub enum GrpcTransportError {
     MessageHubProtoConversion(#[from] MessageHubProtoError),
 }
 
+pub(crate) struct NodeIdInterceptor(Uuid);
+
+impl NodeIdInterceptor {
+    pub(crate) fn new(uuid: Uuid) -> Self {
+        Self(uuid)
+    }
+}
+
+impl Interceptor for NodeIdInterceptor {
+    fn call(&mut self, mut req: Request<()>) -> Result<Request<()>, Status> {
+        let val = MetadataValue::from_str(&self.0.to_string())
```

Comment:
Wouldnt it be possible to perform the conversion to a `MetadataValue` just one time in the `new` method ? this would mean that every interceptor call would just insert into the request our metadata.

Reply:
It could be possible but I see 2 drawbacks:
1. the `new` method should return a `Result` (which is not a big deal)
2. the `call` method has a mut reference to self and the `insert` method requires ownership of the `MetadataValue`, therefore we should clone the value at every insertion (which should approximately have the same overhead of creating a new `MetadataValue` each time).

Btw, your observation makes sense. Let's see what the others think about it